src/main.ts

```diff
@@ -45,7 +47,15 @@ export async function run() {
       auth.configAuthentication(registryUrl, alwaysAuth);
     }
 
-    const matchersPath = path.join(__dirname, '..', '.github');
+    if (cache) {
+      if (!isGhes()) {
+        await restoreCache(cache);
+      } else {
+        throw new Error('Caching is not supported on GHES');
+      }
```

Comment:
Let's simplify the condition
```suggestion
      if (isGhes()) {
        throw new Error('Caching is not supported on GHES');
      } 
      
      await restoreCache(cache);
```

Reply:
Done. Thank you.